pipelinewise/fastsync/commons/split_gzip.py

```diff
@@ -0,0 +1,127 @@
+"""Functions that write chunked gzipped files."""
+import io
+import logging
+import gzip
+
+LOGGER = logging.getLogger(__name__)
+
+DEFAULT_CHUNK_SIZE_MB = 1000
+DEFAULT_MAX_CHUNKS = 20
+
+# Detecting compressed file size at write time is not possible by GzipFile.
+# The data hase to be written into the file first before the actual compression performed.
+# We need to use a good estimate for a text gzip file to split the file and write time.
+EST_COMPR_RATE = 0.12
+
+
+# pylint: disable=W0622
+def open(base_filename, mode='wb', chunk_size_mb=None, max_chunks=None, est_compr_rate=None):
```

Comment:
Why the need for a function? why not use the `SplitGzipFile` constructor do what this function does?

Reply:
This way it's consequent with the built-in python [gzip module](https://github.com/python/cpython/blob/master/Lib/gzip.py) object